airbyte-integrations/connectors/source-klaviyo/source_klaviyo/streams.py

```diff
@@ -210,194 +137,144 @@ def cursor_field(self) -> Union[str, List[str]]:
-
-        return params
-
-    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
-        """
-        Override to determine the latest state after reading the latest record. This typically compared the cursor_field from the latest
-        record and the current state and picks the 'most' recent cursor. This is how a stream's state is determined.
-        Required for incremental.
-        """
-        state_ts = int(current_stream_state.get(self.cursor_field, 0))
-        latest_record = latest_record.get(self.cursor_field)
-
-        if isinstance(latest_record, str):
-            latest_record = datetime.datetime.strptime(latest_record, "%Y-%m-%d %H:%M:%S")
-            latest_record = datetime.datetime.timestamp(latest_record)
-
-        new_value = max(latest_record, state_ts)
-        new_value = min(new_value, self._start_sync)
-        return {self.cursor_field: new_value}
-
-    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
-        """
-        This method should return a Mapping (e.g: dict) containing whatever information required to make paginated requests.
-        This dict is passed to most other methods in this class to help you form headers, request bodies, query params, etc.
-        :param response: the most recent response from the API
-        :return If there is another page in the result, a mapping (e.g: dict) containing info needed to query the next page in the response.
-                If there are no more pages in the result, return None.
-        """
-        decoded_response = response.json()
-        if decoded_response.get("next"):
-            return {"since": decoded_response["next"]}
-
-        return None
-
-
-class ReverseIncrementalKlaviyoStreamV1(KlaviyoStreamV1, ABC):
-    """Base class for all streams that natively incremental but supports desc & asc order"""
-
-    def __init__(self, start_date: Optional[str], **kwargs):
-        super().__init__(**kwargs)
-        self._start_datetime = pendulum.parse(start_date) if start_date else None
-        self._reversed = False
-        self._reached_old_records = False
-        self._low_boundary = None
-
-    @property
-    def state_checkpoint_interval(self) -> Optional[int]:
-        """How often to checkpoint state (i.e: emit a STATE message). By default, return the same value as page_size"""
-        return None if self._reversed else self.page_size
+        starting_point = stream_state.get(self.cursor_field) or self._start_ts
+        records = [
+            record
+            for record in super().read_records(
+                sync_mode=sync_mode, cursor_field=cursor_field, stream_slice=stream_slice, stream_state=stream_state
+            )
+            if starting_point and record[self.cursor_field] > starting_point or not starting_point
+        ]
+        records.sort(key=lambda r: r[self.cursor_field])
```

Comment:
Will this still work for "semi-incremental" case where the API does not allow for sorting? Or this will work "out-of-box" even if it's not sorted here?

Reply:
we can filter each record without sorting, can we?

Reply:
You're definitely right, deleting sorting